### PR TITLE
BUG: freqstr may be parsed incorrectly

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Invalid freq like below doesn'traise `ValueError` properly. The PR makes all the below raise.

```
pd.tseries.frequencies.to_offset('U1')
# <Micro>
pd.tseries.frequencies.to_offset('3U1')
# <3 * Micros>
pd.tseries.frequencies.to_offset('-33-22U')
# <-22 * Micros>
```

### CODE REVIEW

**Comment by jreback** [comment]
might fix / xref to https://github.com/pydata/pandas/issues/8419

**Comment by sinhrks** [comment]
Now ``pd.tseries.frequencies.to_offset('.5S')`` explicitly raises. I'll comment on #8419.

**Comment by jreback** [comment]
@sinhrks merged #13874 so this needs a rebase

**Comment by sinhrks** [comment]
ok, rebased.

File: pandas/tseries/frequencies.py [hunk 1]
```diff
@@ -464,7 +464,14 @@ def to_offset(freq):
         delta = None
         stride_sign = None
         try:
-            for stride, name, _ in opattern.findall(freq):
+            splitted = re.split(opattern, freq)
+            if splitted[-1] != '' and not splitted[-1].isspace():
+                # the last element must be blank
+                raise ValueError('includes garbage characters')
```
Comment (jreback) [inline pandas/tseries/frequencies.py]: maybe a better message here?

**Comment by jreback** [comment]
minor change. lgtm. ping when green.

File: pandas/tseries/frequencies.py (hunk 1 above)
Comment (sinhrks) [inline pandas/tseries/frequencies.py]: this is an internal description which is catched immediately. Users wil see " invalid frequency: xxx" as an error.

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**